marketplace/src/lib.rs

```diff
@@ -83,679 +76,503 @@ pub mod pallet {
+				marketplace.kind = kind;
 				Ok(())
 			})?;
 
-			let event = Event::MarketplaceTypeChanged { marketplace_id, kind };
+			let event = Event::MarketplaceKindSet { marketplace_id, kind };
 			Self::deposit_event(event);
 
 			Ok(().into())
 		}
 
-		#[pallet::weight(T::WeightInfo::set_marketplace_name())]
-		pub fn set_marketplace_name(
+		/// Set the configuration parameters of the marketplace (eg. commission_fee, listing_fee,
+		/// account_list, offchain_data). Must be called by the owner of the marketplace.
+		#[pallet::weight(T::WeightInfo::set_marketplace_configuration())]
+		pub fn set_marketplace_configuration(
 			origin: OriginFor<T>,
 			marketplace_id: MarketplaceId,
-			name: NameVec<T>,
+			commission_fee: ConfigOp<MarketplaceFee<BalanceOf<T>>>,
+			listing_fee: ConfigOp<MarketplaceFee<BalanceOf<T>>>,
+			account_list: ConfigOp<BoundedVec<T::AccountId, T::AccountSizeLimit>>,
+			offchain_data: ConfigOp<BoundedVec<u8, T::OffchainDataLimit>>,
 		) -> DispatchResultWithPostInfo {
-			let caller_id = ensure_signed(origin)?;
+			let who = ensure_signed(origin)?;
 
 			Marketplaces::<T>::try_mutate(marketplace_id, |x| -> DispatchResult {
-				let market_info = x.as_mut().ok_or(Error::<T>::MarketplaceNotFound)?;
-				ensure!(market_info.owner == caller_id, Error::<T>::NotMarketplaceOwner);
-				market_info.name = name.clone();
+				let marketplace = x.as_mut().ok_or(Error::<T>::MarketplaceNotFound)?;
+
+				// Checks
+				ensure!(marketplace.owner == who, Error::<T>::NotTheMarketplaceOwner);
+
+				// Execute
+				config_op_field_exp!(marketplace.commission_fee, commission_fee);
+				config_op_field_exp!(marketplace.listing_fee, listing_fee);
+				config_op_field_exp!(marketplace.account_list, account_list.clone());
+				config_op_field_exp!(marketplace.offchain_data, offchain_data.clone());
 				Ok(())
 			})?;
 
-			let event = Event::MarketplaceNameUpdated { marketplace_id, name };
+			let event = Event::MarketplaceConfigSet {
+				marketplace_id,
+				commission_fee,
+				listing_fee,
+				account_list,
+				offchain_data,
+			};
 			Self::deposit_event(event);
 
 			Ok(().into())
 		}
 
+		/// Set the fee for minting a marketplace if the caller is root.
 		#[pallet::weight(T::WeightInfo::set_marketplace_mint_fee())]
 		pub fn set_marketplace_mint_fee(
 			origin: OriginFor<T>,
-			mint_fee: BalanceOf<T>,
+			fee: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
-
-			MarketplaceMintFee::<T>::put(mint_fee);
-
-			Self::deposit_event(Event::MarketplaceMintFeeUpdated { fee: mint_fee });
+			MarketplaceMintFee::<T>::put(fee);
+			Self::deposit_event(Event::MarketplaceMintFeeSet { fee });
 
 			Ok(().into())
 		}
 
-		#[pallet::weight(T::WeightInfo::set_marketplace_commission_fee())]
-		pub fn set_marketplace_commission_fee(
+		/// Put an NFT on sale on a marketplace.
+		#[pallet::weight(T::WeightInfo::list_nft())]
+		#[transactional]
+		pub fn list_nft(
 			origin: OriginFor<T>,
+			nft_id: NFTId,
```

Comment:
What do you think of reordering this to be: `nft_id`, `marketplace_id` and then `price`.